Error running Scalafix after upgrading from 1.5->1.{6,7,8}

When running Scalafix with the following command options
```
gradle scalafix -PautoConfigureSemanticdb=true -Pscalafix.rules=RemoveUnused
```
I now get the error
```
Running Scalafix on 37 Scala source file(s)
error: Unable to load symbol table: error in opening zip file
```
Running with `--info` throws this
```
> Task :scalafixMain FAILED
Caching disabled for task ':scalafixMain' because:
  Caching has not been enabled for the task
Task ':scalafixMain' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Running Scalafix on 37 Scala source file(s)
error: Unable to load symbol table: error in opening zip file
:scalafixMain (Thread[Execution worker for ':' Thread 6,5,main]) completed. Took 0.206 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':scalafixMain'.
> Errors:
   - A command-line argument parsed incorrectly

```

It's the same for 1.6, 1.7 and 1.8.

However, 1.5 works fine.

My config looks like this
```
rules = [
  DisableSyntax
]

RemoveUnused.patternvars = true
RemoveUnused.imports = true
RemoveUnused.privates = false
RemoveUnused.locals = false

DisableSyntax.noVars = true
DisableSyntax.noThrows = false
DisableSyntax.noNulls = false
DisableSyntax.noReturns = false
DisableSyntax.noWhileLoops = false
DisableSyntax.noAsInstanceOf = false
DisableSyntax.noIsInstanceOf = false
DisableSyntax.noXml = false
DisableSyntax.noDefaultArgs = false
DisableSyntax.noFinalVal = false
DisableSyntax.noFinalize = false
DisableSyntax.noValPatterns = false
DisableSyntax.noUniversalEquality = false
DisableSyntax.noUniversalEqualityMessage = "== and != are unsafe since they allow comparing two unrelated types"
DisableSyntax.regex = []


ExplicitResultTypes.memberKind = [
  Def,
  Val,
  Var
]
ExplicitResultTypes.memberVisibility = [
  Public,
  Protected
]
ExplicitResultTypes.skipSimpleDefinitions = ['Term.Ref', 'Lit', 'Term.New']
ExplicitResultTypes.skipLocalImplicits = true
ExplicitResultTypes.fatalWarnings = false
ExplicitResultTypes.rewriteStructuralTypesToNamedSubclass = true
```

Versions
- Scala 2.12.13
- Oracle Java 1.8.0_251
- Gradle 6.8.3

I wonder if any of the jars you have in the classpath is corrupted. If I add a dependency pointing to a file that is not a jar file or contain invalid content, I get the same error:

```
echo "foo" >  ~/tmp/corrupted.jar
```

`compile files('~/tmp/corrupted.jar')`

```
Running Scalafix on 3 Scala source file(s)
error: Unable to load symbol table: error in opening zip file

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':module-a:scalafixMain'.
> Errors:
   - A command-line argument parsed incorrectly
```

Could you run the scalafix task passing `--debug` and search for `Running Scalafix with the following arguments`. The Scalafix task will log all the parameters passed into Scalafix. Take a look at what is printed after `Classpath: ` and see if you can spot anything apart from jars and class file dirs? If you don't, please take a look at [this](https://stackoverflow.com/questions/7600028/maven-error-in-opening-zip-file-when-running-maven) and see if any of the jars you have in the classpath could be corrupted.

--- a/src/main/groovy/io/github/cosmicsilence/scalafix/ScalaSourceSet.groovy
+++ b/src/main/groovy/io/github/cosmicsilence/scalafix/ScalaSourceSet.groovy
@@ -21,7 +21,15 @@ class ScalaSourceSet {
     }
 
     List<File> getFullClasspath() {
-        return (sourceSet.output.classesDirs + sourceSet.compileClasspath).toList()
+        return classesDirs + jarDependencies
+    }
+
+    private List<File> getClassesDirs() {
+        return sourceSet.output.classesDirs.toList()
+    }
+
+    private List<File> getJarDependencies() {
+        return sourceSet.compileClasspath.toList().findAll { it.name.toLowerCase().endsWith(".jar") }
     }
 
     SourceDirectorySet getScalaSources() {

--- a/src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy
+++ b/src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy
@@ -207,6 +207,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/main")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/main")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -234,6 +235,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/main")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/main")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -262,6 +264,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/test")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/test")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -289,6 +292,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/test")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/test")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -331,6 +335,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/main")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/main")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -358,6 +363,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/main")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/main")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -386,6 +392,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/test")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/test")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -413,6 +420,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/test")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/test")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -443,6 +451,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/foo")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/foo")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -472,6 +481,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/bar")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/bar")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -502,6 +512,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/foo")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/foo")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions == DEFAULT_COMPILER_OPTS
         task.scalaVersion == SCALA_VERSION
         task.rules.get().containsAll(['Foo', 'Bar'])
@@ -531,6 +542,7 @@ class ScalafixPluginTest extends Specification {
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/scala/bar")
         task.classpath.contains(scalaProject.projectDir.path + "/build/classes/java/bar")
         task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.jar") }
+        !task.classpath.find { it.endsWith("scala-library-${SCALA_VERSION}.pom") }
         task.compileOptions.containsAll(DEFAULT_COMPILER_OPTS + "-Yrangepos")
         task.compileOptions.find { it.startsWith("-Xplugin:") }
         task.scalaVersion == SCALA_VERSION
@@ -709,7 +721,8 @@ class ScalafixPluginTest extends Specification {
             }
 
             dependencies {
-                compile "org.scala-lang:scala-library:$scalaVersion"
+                compile group: 'org.scala-lang', name: 'scala-library', version: scalaVersion
+                compile group: 'org.scala-lang', name: 'scala-library', version: scalaVersion, ext: 'pom'
             }
 
             sourceSets {

